### PR TITLE
Fix issue with explicit interface completion and static interface members

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/54005

### CODE REVIEW

File: src/EditorFeatures/CSharpTest/Completion/CompletionProviders/ExplicitInterfaceTypeCompletionProviderTests.cs [hunk 1]
```diff
@@ -358,4 +356,23 @@ class C<T> : I<T>
...
     }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/54005")]
+    public async Task TestWithStaticKeyword()
+    {
+        var markup = """
+            interface I1
+            {
+
+                static abstract void M1();
```
Comment (ToddGrun) [inline src/EditorFeatures/CSharpTest/Completion/CompletionProviders/ExplicitInterfaceTypeCompletionProviderTests.cs:367]: >static [](http://example.com/codeflow?start=16&length=6)

nit: empty line above

Comment (ToddGrun) [inline src/EditorFeatures/CSharpTest/Completion/CompletionProviders/ExplicitInterfaceTypeCompletionProviderTests.cs:367]: >static [](http://example.com/codeflow?start=16&length=6)

nit: empty line above

File: src/EditorFeatures/CSharpTest/Completion/CompletionProviders/ExplicitInterfaceTypeCompletionProviderTests.cs [hunk 2]
```diff
@@ -358,4 +356,23 @@ class C<T> : I<T>
...
         await VerifyItemExistsAsync(markup, "I", displayTextSuffix: "<>");
     }
+
+    [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/54005")]
+    public async Task TestWithStaticKeyword()
+    {
+        var markup = """
+            interface I1
+            {
+
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/Completion/CompletionProviders/ExplicitInterfaceTypeCompletionProviderTests.cs:366]:
```suggestion
```

**Review by ToddGrun — APPROVED** [review]
:shipit:

**Pull request merged**